cpp/src/io/fst/lookup_tables.cuh

```diff
@@ -179,6 +179,72 @@ class SingleSymbolSmemLUT {
   }
 };
 
+/**
+ * @brief A simple symbol group lookup wrapper that uses a simple function object to
+ * retrieve the symbol group id for a symbol.
+ *
+ * @tparam SymbolGroupLookupOpT The function object type to return the symbol group for a given
+ * symbol
+ */
+template <typename SymbolGroupLookupOpT>
+class SymbolGroupLookupOp {
+ private:
+  struct _TempStorage {};
+
+ public:
+  using TempStorage = cub::Uninitialized<_TempStorage>;
+
+  struct KernelParameter {
+    using LookupTableT = SymbolGroupLookupOp<SymbolGroupLookupOpT>;
```

Comment:
this is used in Dfa?

Reply:
Exactly. The DFA uses this member type to instantiate the class that performs the symbol group id lookups.